windows_service/datadog_checks/windows_service/windows_service.py

```diff
@@ -3,14 +3,134 @@
 # Licensed under a 3-clause BSD style license (see LICENSE)
 import re
 
+import pywintypes
 import win32service
-from six import iteritems
 
 from datadog_checks.base import AgentCheck
 
 SERVICE_PATTERN_FLAGS = re.IGNORECASE
 
 
+class ServiceFilter(object):
+    def __init__(self, name=None, startup_type=None):
+        self.name = name
+        self.startup_type = startup_type
+
+        self._init_patterns()
+
+    def _init_patterns(self):
+        if self.name is not None:
+            self._name_re = re.compile(self.name, SERVICE_PATTERN_FLAGS)
+        if self.startup_type is not None:
+            self._startup_type_re = re.compile(self.startup_type, SERVICE_PATTERN_FLAGS)
+
+    def match(self, service_view):
+        if self.name is not None:
+            if not self._name_re.match(service_view.name):
+                return False
+        if self.startup_type is not None:
+            if not self._startup_type_re.match(service_view.startup_type_string()):
+                return False
+        return True
```

Comment:
In yaml can be OR condition be specified

Reply:
AND
```
services:
  - name: EventLog
    startup_type: automatic
```

OR
```
services:
  - name: EventLog
  - startup_type: automatic
```

Reply:
Right, I realized it after I have sent it. Agree.